File: output/GPT3.5-Turbo/xCodeEval_4000/C++/C/b1690c5cf357a70cfbba1c62bf1c2c6f.c
#include <stdio.h>
#include <string.h>

#define MAXN 1000100

int getq()
{
    char buf[50];
    int ret=0;
    scanf("%s",buf);
    for(int i=strlen(buf)-1;i>=0;i--)ret|=1<<(buf[i]-'a');
    return ret;
}

int main()
{
    char S[MAXN];
    int n,m;
    int Q[10010];
    int ans[10010];
    int Qs[10010];
    int Qs_size = 0;
    scanf("%s",S+1);
    n=strlen(S+1);
    for(int i=1;i<=n;i++)S[i]-='a';
    scanf("%d",&m);
    for(int i=1;i<=m;i++){
        int q = getq();
        int found = 0;
        for(int j=0;j<Qs_size;j++){
            if(Qs[j] == q){
                ans[i] = ans[j];
                found = 1;
                break;
            }
        }
        if(!found){
            Qs[Qs_size] = q;
            ans[i] = 0;
            for(int j=1;j<=n;j++){
                int x=S[j];
                if(!vis[x])stk[++nn]=x,vis[x]=1;
                else
                {
                    int k;
                    for(k=1;;k++)if(stk[k]==x)break;
                    for(;k<nn;k++)stk[k]=stk[k+1];
                    stk[nn]=x;
                }
                int t=0;
                for(int k=nn;k;k--)
                {
                    t|=1<<stk[k];
                    if(j<n&&(t&(1<<S[j+1])))break;
                    if(Qs_size > 0 && Qs[Qs_size-1] == t){
                        ans[i]++;
                    }
                }
            }
            Qs_size++;
        }
    }
    for(int i=1;i<=m;i++)printf("%d\n",ans[i]);
    return 0;
}

File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Go/b1690c5cf357a70cfbba1c62bf1c2c6f.go
package main

import (
	"fmt"
	"strings"
)

func getq() int {
	var buf string
	fmt.Scan(&buf)
	ret := 0
	for i := len(buf) - 1; i >= 0; i-- {
		ret |= 1 << (buf[i] - 'a')
	}
	return ret
}

func main() {
	var S string
	fmt.Scan(&S)
	n := len(S)
	S = strings.ToLower(S)
	var m int
	fmt.Scan(&m)
	Q := make([]int, m)
	Qs := make(map[int]bool)
	ans := make(map[int]int)
	for i := 0; i < m; i++ {
		Q[i] = getq()
		Qs[Q[i]] = true
	}
	stk := make([]int, 27)
	nn := 0
	vis := make([]bool, 26)
	for i := 0; i < n; i++ {
		x := int(S[i] - 'a')
		if !vis[x] {
			nn++
			stk[nn] = x
			vis[x] = true
		} else {
			j := 0
			for ; j < nn; j++ {
				if stk[j] == x {
					break
				}
			}
			for ; j < nn; j++ {
				stk[j] = stk[j+1]
			}
			stk[nn] = x
		}
		t := 0
		for j := nn; j > 0; j-- {
			t |= 1 << stk[j]
			if i < n-1 && (t&(1<<(S[i+1]-'a'))) != 0 {
				break
			}
			if _, ok := Qs[t]; ok {
				ans[t]++
			}
		}
	}
	for i := 0; i < m; i++ {
		fmt.Println(ans[Q[i]])
	}
}
